pennylane/templates/subroutines/qdrift.py

```diff
@@ -0,0 +1,257 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Contains template for QDrift subroutine."""
+
+import pennylane as qml
+from pennylane.operation import Operation
+from pennylane.math import requires_grad, unwrap
+from pennylane.ops import Sum, SProd, Hamiltonian
+
+
+@qml.QueuingManager.stop_recording()
+def _sample_decomposition(coeffs, ops, time, n=1, seed=None):
+    """Generate the randomly sampled decomposition
+
+    Args:
+        coeffs (array): the coefficients of the operations from each term in the Hamiltonian
+        ops (list[~.Operator]): the normalized operations from each term in the Hamiltonian
+        time (float): time to evolve under the target Hamiltonian
+        n (int): number of samples in the product. defaults to 1
```

Comment:
[not blocker]: I'm still not sure it makes sense to put n = 1. It's quite different from the n= 1 in Trotter

Reply:
That's a fair point, I think 1 is a good default value in general, in my opinion they should read the docs to understand what the argument means.

Reply:
Added